File: LL/src/Core/Domain/Models/MarketPlaces/MarketPlaceOrder.cs
using Domain.Models.Items;

namespace Domain.Models.MarketPlaces;

public enum MarketPlaceTradeSource
{
    SellListing = 0,
    BuyOrder = 1
}

public class MarketPlaceOrder
{
    public Guid Id { get; set; }
    public Guid SellerId { get; set; }
    public Guid BuyerId { get; set; }
    public string ItemBaseId { get; set; } = string.Empty;
    public ItemBase ItemBase { get; set; } = null!;
    public Guid? ItemInstanceId { get; set; }
    public int Quantity { get; set; }
    public long UnitPrice { get; set; }
    public long TotalPrice { get; set; }
    public long SellerFee { get; set; }
    public MarketPlaceTradeSource Source { get; set; }
    public DateTimeOffset PurchasedAt { get; set; }
}

File: LL/src/Presentation/ll/src/app/shared/models/Dtos/market-place/market-place-order.ts
import { ItemBase } from '../../item';

export interface MarketPlaceOrder {
  id: string;
  sellerId: string;
  buyerId: string;
  itemBaseId: string;
  itemBase: ItemBase;
  itemInstanceId: string | null;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  sellerFee: number;
  source: number | string;
  purchasedAt: Date;
}
